testsuite/forge/src/backend/k8s_deployer/deployer.rs

```diff
@@ -0,0 +1,444 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    FORGE_DEPLOYER_IMAGE_TAG, FORGE_DEPLOYER_SERVICE_ACCOUNT_NAME,
+    FORGE_DEPLOYER_VALUES_ENV_VAR_NAME, FORGE_INDEXER_DEPLOYER_DOCKER_IMAGE_REPO,
+    FORGE_TESTNET_DEPLOYER_DOCKER_IMAGE_REPO,
+};
+use crate::{maybe_create_k8s_resource, K8sApi, ReadWrite, Result};
+use k8s_openapi::api::{
+    batch::v1::Job,
+    core::v1::{ConfigMap, Namespace, ServiceAccount},
+    rbac::v1::RoleBinding,
+};
+use kube::{
+    api::{ObjectMeta, PostParams},
+    ResourceExt,
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::BTreeMap, fmt, sync::Arc};
+
+/// These are the values that the forge deployer needs to deploy the forge components to the k8s cluster.
+/// There are global values such as profile, era, and namespace as well as component-specific values
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ForgeDeployerValues {
+    pub profile: String,
+    pub era: String,
+    pub namespace: String,
+
+    // component specific values
+    // TODO: add an options reference. Ideally this customization is almost always optional and instead handled by the profiles
+    pub indexer_grpc_values: Option<serde_json::Value>,
+    pub indexer_processor_values: Option<serde_json::Value>,
+}
+
+/// The ForgeDeployerManager is responsible for managing the lifecycle of forge deployers, wihch deploy the
+/// forge components to the k8s cluster.
+pub struct ForgeDeployerManager {
+    // all the k8s APIs we need. Specifying each API separately allows for easier testing
+    pub jobs_api: Arc<dyn ReadWrite<Job>>,
+    pub config_maps_api: Arc<dyn ReadWrite<ConfigMap>>,
+    pub namespace_api: Arc<dyn ReadWrite<Namespace>>,
+    pub serviceaccount_api: Arc<dyn ReadWrite<ServiceAccount>>,
+    pub rolebinding_api: Arc<dyn ReadWrite<RoleBinding>>,
```

Comment:
You can replace all these as mentioned above with an on demand kube_api! call :D